contracts/src/v0.1/mocks/L2RequestResponseConsumerMock.sol

```diff
@@ -0,0 +1,176 @@
+import "../RequestResponseConsumerBase.sol";
+import "../interfaces/IL2Endpoint.sol";
+
+contract L2RequestResponseConsumerMock is
+    RequestResponseConsumerFulfillUint128,
+    RequestResponseConsumerFulfillInt256,
+    RequestResponseConsumerFulfillBool,
+    RequestResponseConsumerFulfillString,
+    RequestResponseConsumerFulfillBytes32,
+    RequestResponseConsumerFulfillBytes
+{
+    using Orakl for Orakl.Request;
+    uint128 public sResponseUint128;
+    int256 public sResponseInt256;
+    bool public sResponseBool;
+    string public sResponseString;
+    bytes32 public sResponseBytes32;
+    bytes public sResponseBytes;
+    address private sOwner;
+
+    IL2Endpoint L2ENDPOINT;
+
+    error OnlyOwner(address notOwner);
+
+    modifier onlyOwner() {
+        if (msg.sender != sOwner) {
+            revert OnlyOwner(msg.sender);
+        }
+        _;
+    }
+
+    constructor(address l2Endpoint) RequestResponseConsumerBase(l2Endpoint) {
+        sOwner = msg.sender;
+        L2ENDPOINT = IL2Endpoint(l2Endpoint);
+    }
+
+    // Receive remaining payment from requestRandomWordsPayment
+    receive() external payable {}
+
+    //request for uint128
+    function requestDataUint128(
+        uint64 accId,
+        uint32 callbackGasLimit,
+        uint8 numSubmission
+    ) public onlyOwner returns (uint256 requestId) {
+        bytes32 jobId = keccak256(abi.encodePacked("uint128"));
+        Orakl.Request memory req = buildRequest(jobId);
+        //change here for your expected data
+        req.add(
+            "get",
+            "https://min-api.cryptocompare.com/data/pricemultifull?fsyms=KLAY&tsyms=USD"
+        );
+        req.add("path", "RAW,KLAY,USD,PRICE");
+        req.add("pow10", "8");
+        requestId = L2ENDPOINT.requestData(req, callbackGasLimit, accId, numSubmission);
+    }
+
+    // request for int256
+    function requestDataInt256(
+        uint64 accId,
+        uint32 callbackGasLimit,
+        uint8 numSubmission
+    ) public onlyOwner returns (uint256 requestId) {
+        bytes32 jobId = keccak256(abi.encodePacked("int256"));
+        Orakl.Request memory req = buildRequest(jobId);
+        //change here for your expected data
+        req.add(
+            "get",
+            "https://min-api.cryptocompare.com/data/pricemultifull?fsyms=KLAY&tsyms=USD"
+        );
+        req.add("path", "RAW,KLAY,USD,PRICE");
+        req.add("pow10", "8");
+
+        requestId = L2ENDPOINT.requestData(req, callbackGasLimit, accId, numSubmission);
+    }
+
+    // request for bool
+    function requestDataBool(
+        uint64 accId,
+        uint32 callbackGasLimit,
+        uint8 numSubmission
+    ) public onlyOwner returns (uint256 requestId) {
+        bytes32 jobId = keccak256(abi.encodePacked("bool"));
+        Orakl.Request memory req = buildRequest(jobId);
+        req.add(
+            "get",
+            "https://min-api.cryptocompare.com/data/pricemultifull?fsyms=KLAY&tsyms=USD"
```

Comment:
@KelvinThai We should be careful about the API endpoints that we call in this contract because for most of them the conversion does not not make sense. In this example we request bool data and call KLAY/USD API endpoint. However, the bigger issue is with the javascript tests which do not process the request that is defined in this smart contract, but rather supply whatever is expected and then re-read from the consumer contract. I think it is okay to do it like this if we add explanation to the code so people are not surprise when they try to understand it.